### DISCUSSION TITLE
getting ValueError: 'vertices' must be a 2D list or array with shape Nx2

### DISCUSSION THREAD

**Discussion by samahwaleed** [networkx/networkx#5241] [Q&A]
I have a dictionary where is a key is the node name and the value is the node position. the dictionary `(dicts)` looks like this:
```
{0: array([ 360, 259, 364, 259, 364, 261, 360, 261]), 
 1: array([ 380, 258, 385, 258, 385, 262, 380, 262]), 
 2: array([ 365, 258, 370, 258, 370, 262, 365, 262]), 
 3: array([ 386, 256, 393, 256,393, 260,386, 260])}
```

I am trying to plot a graph using networkx but I got error:

> ValueError: 'vertices' must be a 2D list or array with shape Nx2

My code:

```
 g = torch_geometric.utils.to_networkx(data, to_undirected=True)
  plt.figure(len(data))
  nx.draw(g, pos=dicts, with_labels=True)
  plt.show()
```

**Comment by dschult** [comment]
Is this a question for `torch_geometric`?
Try to track down the course of the error.

**Comment by rossbar** [comment]
It's not clear how the values in the `dicts` dictionary are supposed to be interpreted. The drawing functions in networkx expect the `pos` dictionary keyed by node where the values represent the 2D coordinates of nodes, e.g.

```python
pos = {
    0: (0, 0),  # Node 0 located at the origin (0, 0)
    1: (1, 1),  # Node 1 at cartesian position (1, 1)
}
```
etc.
